src/plans/stepsize.jl

```diff
@@ -660,30 +660,604 @@ If ``$(_tex(:norm, "X_k"; index = "p_k")) ≤ αω_{k-1}``, then let
+    else
+        #(R5)
+        if (c.df * (a.t - c.t) < 0)
+            a, b = c, a
+        elseif (a.df * (b.t - a.t) < 0)
+            a, b = a, c
+        else
+            a, b = c, b
+        end
+    end
+    return a, b
+end
+
+"""
+    cubic_polynomial_argmin(a::UnivariateTriple, b::UnivariateTriple; warn::Bool = true)
+
+Returns the local minimizer of the cubic polynomial ``p`` with ``p(a.t)=a.f``, ``p(b.t)=b.f``,
+``p'(a.t)=a.df``, ``p'(b.t)=b.df``.
+
+# Input
+* `a::UnivariateTriple{R}`: triple of bracket value `a`
+* `b::UnivariateTriple{R}`: triple bracket value `b`
+
+# Keyword arguments
+* `warn::Bool`: Boolean value if warnings should be displayed
+"""
+function cubic_polynomial_argmin(a::UnivariateTriple{R}, b::UnivariateTriple{R}; warn::Bool = true) where {R}
+    (a.f > b.f && warn) && @warn "value bracket condition not met."
+    (a.df * (b.t - a.t) > 0 && warn) && @warn "derivative bracket condition not met."
+
+    Δ = b.t - a.t
+    v = a.df + b.df - 3 * (b.f - a.f) / Δ
+    discriminant = v^2 - a.df * b.df
+    #negative discriminants only occur with roundoff errors at 0
+    discriminant = max(discriminant, 0.0)
+    w = sign(Δ) * sqrt(discriminant)
+    denom_a = a.df + v - w
+    denom_b = b.df + v + w
+    if (denom_a > denom_b)
+        return a.t + Δ * a.df / denom_a
+    else
+        return b.t - Δ * b.df / denom_b
+    end
+end
+
+"""
+    secant(a::UnivariateTriple, b::UnivariateTriple)
+
+Returns the extremal of the quadratic polynomial ``p`` with
+``p'(a.t)=a.df``, ``p'(b.t)=b.df``.
+
+# Input
+* `a::UnivariateTriple{R}`: triple of bracket value `a`
+* `b::UnivariateTriple{R}`: triple bracket value `b`
+"""
+function secant(a::UnivariateTriple{R}, b::UnivariateTriple{R}) where {R}
+    return (a.t * b.df - b.t * a.df) / (b.df - a.df)
+end
+
+"""
+    cubic_stepsize_update_step(a::Real, b::Real, c::Real, τ::Real)
+
+
+Step function to determine the stepsize update `c` described in
+[Hager:1989](@cite).
+
+# Input
+* `a::Real`: first value of the bracket
+* `b::Real`: second value of the bracket
+* `c::Real`: update value
+* `τ::Real`: minimal step tolerance
+"""
+function cubic_stepsize_update_step(a::Real, b::Real, c::Real, τ::Real)
+    y = min(a, b)
+    z = max(a, b)
+    if (y + τ ≤ c && c ≤ z - τ)
+        return c
+    end
+    if (c > (a + b) / 2)
+        return max(z - τ, (a + b) / 2)
+    else
+        return min(y + τ, (a + b) / 2)
+    end
+end
+
+"""
+Get the `UnivariateTriple` related to the step with
+stepsize ``τ`` from ``p`` in direction ``η``.
+
+# Input
+* `a::Real`: first value of the bracket
+* `b::Real`: second value of the bracket
+* `c::Real`: update value
+* `τ::Real`: minimal step tolerance
+"""
+function get_univariate_triple!(mp, cbls, p, η, t)
+    M = get_manifold(mp)
+    cbls.last_stepsize = t
+    ManifoldsBase.retract_fused!(M, cbls.candidate_point, p, η, t, cbls.retraction_method)
+    Tη = vector_transport_to(M, p, η, cbls.candidate_point, cbls.vector_transport_method)
```

Comment:
It's usually faster to use the in-place variant of vector transport and have the temporary value of `Tη` stored in `cbls`.